shared/contract-utils/test/checkpointing.js

```diff
@@ -0,0 +1,122 @@
+const { assertRevert } = require('@aragon/test-helpers/assertThrow')
+
+const CheckpointingWrapper = artifacts.require('CheckpointingWrapper')
+
+contract('Checkpointing', () => {
+  let checkpointing
+
+  const generateRandomTest = size => {
+    const rand = () => parseInt(10000 * Math.random())
+
+    let values = []
+    let expects = []
+
+    for (let i = 0; i < size; i++) {
+      const prev = values[i - 1] || { t: 0, v: 0 }
+      const t = 1 + prev.t + rand()
+      const v = rand()
+      values.push({ t, v })
+
+      expects.push({ t: t - 1, v: prev.v })
+      expects.push({ t, v })
+    }
+
+    return {
+      values,
+      expects,
+      size,
+    }
+  }
+
+  const tests = [
+    {
+      description: 'odd number of checkpoints',
```

Comment:
We should also add these cases:

- No checkpoints
- One check point
- Two check points